internal/project/documentregistry.go

```diff
@@ -89,12 +89,14 @@ func (r *DocumentRegistry) getDocumentWorker(
 	key registryKey,
 ) *ast.SourceFile {
 	scriptTarget := core.IfElse(scriptInfo.scriptKind == core.ScriptKindJSON, core.ScriptTargetJSON, compilerOptions.GetEmitScriptTarget())
+	scriptInfoVersion := scriptInfo.Version()
+	scriptInfoText := scriptInfo.Text()
```

Comment:
Technically this should be one atomic access

Reply:
Do you want to address this now or later?

Reply:
I just tried protecting all the ScriptInfo fields with a mutex and accessing correlated fields within the same lock, but that had a ~20% penalty on project initialization